unixfs/src/walk.rs

```diff
@@ -483,16 +490,23 @@ impl InnerEntry {
     }
 
     pub fn as_entry(&self) -> Entry<'_> {
-        use InnerKind::*;
         match &self.kind {
-            RootDirectory(cid) | BucketAtRoot(cid) => {
-                Entry::RootDirectory(cid, &self.path, &self.metadata)
+            InnerKind::Bucket(cid) => Entry::Bucket(cid, &self.path),
+            InnerKind::Directory(cid) => {
+                Entry::Metadata(MetadataEntry::Directory(cid, &self.path, &self.metadata))
+            }
+            InnerKind::File(cid, _, sz) => {
+                Entry::Metadata(MetadataEntry::File(cid, &self.path, &self.metadata, *sz))
+            }
+            InnerKind::RootBucket(cid) => {
+                Entry::Metadata(MetadataEntry::Directory(cid, &self.path, &self.metadata))
+            }
+            InnerKind::RootDirectory(cid) | InnerKind::BucketAtRoot(cid) => Entry::Metadata(
+                MetadataEntry::RootDirectory(cid, &self.path, &self.metadata),
+            ),
+            InnerKind::Symlink(cid) => {
+                Entry::Metadata(MetadataEntry::Symlink(cid, &self.path, &self.metadata))
             }
-            RootBucket(cid) => Entry::Directory(cid, &self.path, &self.metadata),
-            Bucket(cid) => Entry::Bucket(cid, &self.path),
-            Directory(cid) => Entry::Directory(cid, &self.path, &self.metadata),
-            File(cid, _, sz) => Entry::File(cid, &self.path, &self.metadata, *sz),
-            Symlink(cid) => Entry::Symlink(cid, &self.path, &self.metadata),
         }
     }
```

Comment:
In hindsight this seems "really obvious" change, great idea!

Thinking of this naming or meaningwise the difference between the cases might be more on the `Entry::Metadata` meaning "is standalone entry [which might have non-empty metadata]" and `Entry::Bucket` meaning "is continuation [or 'just more links']" but as I think the doc comments on `Entry` variants already cover this detail nicely so no need to change this *right now*.